packages/zipkin/index.d.ts

```diff
@@ -287,8 +287,17 @@ declare namespace zipkin {
     letContext<V>(ctx: TraceId, callback: () => V): V;
   }
 
+  type RequestZipkinHeaders<T = any, H = any> = T & {
+    headers: H & {
+      ['X-B3-TraceId']: string;
+      ['X-B3-SpanId']: string;
+      ['X-B3-ParentSpanId']?: string;
+      ['X-B3-Sampled']?: '1' | '0';
```

Comment:
ps and assuming this is server-side:
* we often leniently parse 'true' and 'false' for old clients, as they used to send this.
* we'll want to update this for 'b3' header once that chore is done.

Reply:
also, I think this is missing `['X-B3-Flags']?: '1' | '0';` eventhough '0' is redundant and should not be sent.. some might send it. Flag 1 is "debug"

Reply:
Hi @adriancole, nice to meet you, thank you very much for your suggestions. I paste the content of the current **request.js** file below:
```
const HttpHeaders = require('./httpHeaders');

function appendZipkinHeaders(req, traceId) {
  const headers = req.headers || {};
  headers[HttpHeaders.TraceId] = traceId.traceId;
  headers[HttpHeaders.SpanId] = traceId.spanId;

  traceId._parentId.ifPresent(psid => {
    headers[HttpHeaders.ParentSpanId] = psid;
  });
  traceId.sampled.ifPresent(sampled => {
    headers[HttpHeaders.Sampled] = sampled ? '1' : '0';
  });

  return headers;
}

function addZipkinHeaders(req, traceId) {
  const headers = appendZipkinHeaders(req, traceId);
  return Object.assign({}, req, {headers});
}

module.exports = {
  addZipkinHeaders
};
```
When I read the code above, I saw only HttpHeaders.TraceId, HttpHeaders.SpanId, HttpHeaders.ParentSpanId, HttpHeaders.Sampled and headers[HttpHeaders.Sampled] is either '1' or '0'.

> we often leniently parse 'true' and 'false' for old clients, as they used to send this.
It might not affect old clients, because the index.d.ts file is only for typescript projects to use type inference at compile time, so it does not influence old-clients runtime.

>  I think this is missing ['X-B3-Flags']?: '1' | '0';
I couldn't find ['X-B3-Flags'] on the source code above, but if it is **optional(with ?)** and not presented on the source code, we might ignore it.

My response is only based on my limited knowledge about the project; please clarify me if I miss-understood something.